salt/metalk8s/addons/dex/deployed/chart.sls

```diff
@@ -0,0 +1,139 @@
+  labels: {app.kubernetes.io/instance: dex, app.kubernetes.io/managed-by: salt, app.kubernetes.io/name: dex,
+    app.kubernetes.io/part-of: metalk8s, app.kubernetes.io/version: 2.19.0, helm.sh/chart: dex-2.4.0,
+    heritage: metalk8s}
+  name: dex
+  namespace: metalk8s-auth
+spec:
+  ports:
+  - {name: http, port: 32000, targetPort: http}
+  selector: {app.kubernetes.io/instance: dex, app.kubernetes.io/name: dex}
+  sessionAffinity: None
+  type: ClusterIP
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels: {app.kubernetes.io/component: dex, app.kubernetes.io/instance: dex, app.kubernetes.io/managed-by: salt,
+    app.kubernetes.io/name: dex, app.kubernetes.io/part-of: metalk8s, app.kubernetes.io/version: 2.19.0,
+    helm.sh/chart: dex-2.4.0, heritage: metalk8s}
+  name: dex
+  namespace: metalk8s-auth
+spec:
+  replicas: 2
+  selector:
+    matchLabels: {app.kubernetes.io/component: dex, app.kubernetes.io/instance: dex,
+      app.kubernetes.io/name: dex}
+  strategy:
+    rollingUpdate: {maxSurge: 0, maxUnavailable: 1}
+    type: RollingUpdate
+  template:
+    metadata:
+      annotations: {checksum/config: acd60ed0ce19aa32f3c022c95c8499491e0218901023741016ffdde9aeeda8cd}
+      labels: {app.kubernetes.io/component: dex, app.kubernetes.io/instance: dex,
+        app.kubernetes.io/name: dex}
+    spec:
+      containers:
+      - command: [/usr/local/bin/dex, serve, /etc/dex/cfg/config.yaml]
+        env: []
+        image: '{%- endraw -%}{{ build_image_name("dex", False) }}{%- raw -%}:v2.19.0'
+        imagePullPolicy: IfNotPresent
+        name: main
+        ports:
+        - {containerPort: 5556, name: http, protocol: TCP}
+        resources: null
+        volumeMounts:
+        - {mountPath: /etc/dex/cfg, name: config}
+      nodeSelector: {node-role.kubernetes.io/infra: ''}
+      serviceAccountName: dex
+      tolerations:
+      - {effect: NoSchedule, key: node-role.kubernetes.io/bootstrap, operator: Exists}
+      - {effect: NoSchedule, key: node-role.kubernetes.io/infra, operator: Exists}
+      volumes:
+      - name: config
+        secret:
+          defaultMode: 420
+          items:
+          - {key: config.yaml, path: config.yaml}
+          secretName: dex
+---
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  annotations: {kubernetes.io/ingress.class: nginx-control-plane, nginx.ingress.kubernetes.io/backend-protocol: HTTPS,
+    nginx.ingress.kubernetes.io/rewrite-target: /$2, nginx.ingress.kubernetes.io/use-regex: 'true'}
+  labels: {app.kubernetes.io/instance: dex, app.kubernetes.io/managed-by: salt, app.kubernetes.io/name: dex,
+    app.kubernetes.io/part-of: metalk8s, app.kubernetes.io/version: 2.19.0, helm.sh/chart: dex-2.4.0,
+    heritage: metalk8s}
+  name: dex
+  namespace: metalk8s-auth
+spec:
+  rules:
+  - host: null
+    http:
+      paths:
+      - backend: {serviceName: dex, servicePort: 32000}
+        path: /oidc(/|$)(.*)
+  tls:
+  - {hosts: null, secretName: dex-tls-secret}
```

Comment:
Where is this created?

Reply:
Working on this, I am going to add a `secret` that would hold `cert` and `key`